ibis/config.py

```diff
@@ -200,7 +202,17 @@ def __getattr__(self, key):
         try:
             v = self.d[key]
         except KeyError as e:
-            raise AttributeError(*e.args)
+            # See if the options doesn't exist because it's from a backend
```

Comment:
In what way it breaks if we didn't do this here? It seems to me to be a reasonable behavior to error out if user tries to access an option defined by a backend without importing that backend first...

Reply:
This change makes this code work:
```python
import ibis
ibis.options.impala.temp_table = 'foo'
ibis.impala.connect(...)
```
If we break here, and make users write the next code, then wouldn't be needed (some other changes would be needed to this PR for it to work):
```python
import ibis
import ibis.backends.impala

ibis.options.impala.temp_table = 'foo'
ibis.impala.connect(...)
```

Reply:
yes i think thi sis better

Reply:
kk SGTM

Reply:
Should we simply use something like:
```python
import ibis.backends.impala

ibis.backends.impala.set_option('temp_table', 'foo')
ibis.backends.impala.connect(...)
```
or the syntax I proposed in the past
```python
import ibis

ibis.backends.impala.connect(..., temp_table='foo')
# or
ibis.backends.impala.connect(..., options={'temp_table': 'foo'})
```


I think `ibis.<backend-name>` is already cumbersome, and caused several problems. And IMHO this is becoming even more cumbersome, to have to import an unrelated module to make the above code work is far from what I'd expect to have to write as a user.

Reply:
i prefer

```
import ibis.backends.impala 
ibis.backends.impala.set_option('temp_table', 'foo') 
ibis.backends.impala.connect(...)```
```

connect has nothing to do with setting the options of the backend itself (and may actually have multiple invokations). typing these together is not great.

all that said this can be a followup.